app/api/discount_codes.py

```diff
@@ -108,6 +108,16 @@ class DiscountCodeDetail(ResourceDetail):
     """
 
     def before_get(self, args, kwargs):
+        if kwargs.get('ticket_id') and not has_access('is_coorganizer'):
+            raise UnprocessableEntity({'source': ''},
+                                      "Please verify your permission. You\
+                                      must have coorganizer privileges to\
+                                      view ticket discount code details")
+        if kwargs.get('event_id') and not has_access('is_admin'):
+            raise UnprocessableEntity({'source': ''},
+                                      "Please verify your permission. You must be admin to view event\
+                                      discount code details")
+
```

Comment:
There is a lot of redundancy in the method `before_get` i.e. we are checking for permissions too many times. Please go through the method again and I'm sure you'll find out the redundancy. 
If you've added  this above, I think you can safely delete other checks for permissions below this now.